crates/ty_project/src/metadata/options.rs

```diff
@@ -550,3 +550,20 @@ impl OptionDiagnostic {
         }
     }
 }
+
+/// This is a wrapper for options that actually get loaded from configuration files
+/// and the CLI, which also includes a `config_file_override` option that overrides
+/// default configuration discovery with an explicitly-provided path to a configuration file
+pub struct MetaOptions {
```

Comment:
I agree, this name isn't great :)

How about:

* `ConfigurationOverrides`
* `OptionOverrides`
* `ProjectOptionsOverrides`. I think I like this best because these options override the project's options.

We could then rename `cli_options` to just `options` because I forsee that we'll need this for the LSP too